src/shared/database/amendments.ts

```diff
@@ -10,77 +14,163 @@ const log = logger({ name: 'amendments' })
-        if (name2) {
-          amendmentNames.set(name2[1], true)
-        }
-      }
+    log.info(`Updating amendment info for ${network}...`)
+    const client = new Client(url)
+    await client.connect()
+    const featureResponse: FeatureAllResponse = await client.request({
+      command: 'feature',
     })
-    return amendmentNames
-  } catch (err) {
-    log.error('Error getting amendment names', err)
-    return null
+
+    const features = featureResponse.result.features
+
+    for (const id of Object.keys(features)) {
+      amendmentIDs.set(id, {
+        name: features[id].name,
+        deprecated: RETIRED_AMENDMENTS.includes(features[id].name),
+      })
+    }
+
+    await client.disconnect()
+
+    log.info(`Finished updating amendment info for ${network}...`)
+  } catch (error) {
+    log.error(
+      `Failed to update amendment info for ${network} due to error: ${String(
+        error,
+      )}`,
+    )
   }
 }
 
 /**
- * Extracts Amendment ID from Amendment name inside a buffer.
+ * Fetch an amendment info from a network.
  *
- * @param buffer -- The buffer containing the amendment name.
+ * @param amendment_id - The id of the amendment to fetch.
+ * @param url - The Faucet URL of the network.
  *
- * @returns The amendment ID string.
+ * @returns True if the amendment is found, false otherwise.
  */
-function sha512Half(buffer: Buffer): string {
-  return createHash('sha512')
-    .update(buffer)
-    .digest('hex')
-    .toUpperCase()
-    .slice(0, 64)
+async function fetchSingleAmendment(
+  amendment_id: string,
+  url: string,
+): Promise<boolean> {
+  try {
+    log.info(`Updating amendment info for ${amendment_id}...`)
+    const client = new Client(url)
+    await client.connect()
+    const featureResponse: FeatureOneResponse | ErrorResponse =
+      await client.request({
+        command: 'feature',
+        feature: amendment_id,
+      })
+
+    if ('result' in featureResponse) {
+      const feature = featureResponse.result[amendment_id]
+      amendmentIDs.set(amendment_id, {
+        name: feature.name,
+        deprecated: RETIRED_AMENDMENTS.includes(feature.name),
+      })
+      return true
+    }
+
+    await client.disconnect()
+    return false
+  } catch (_error) {
+    return false
+  }
 }
 
 /**
- * Maps the id of Amendments to its corresponding names.
+ * Search for a single amendment across networks.
+ *
+ * @param amendment_id - The id of the amendment to search.
  *
  * @returns Void.
  */
-async function nameOfAmendmentID(): Promise<void> {
-  // The Amendment ID is the hash of the Amendment name
-  const amendmentNames = await fetchAmendmentNames()
-  if (amendmentNames !== null) {
-    amendmentNames.forEach((deprecated, name) => {
-      amendmentIDs.set(sha512Half(Buffer.from(name, 'ascii')), {
-        name,
-        deprecated,
-      })
+async function findSingleAmendment(amendment_id: string): Promise<void> {
+  for (const url of Object.values(NETWORKS_HOSTS)) {
```

Comment:
I think a function with a name starting with `find` should return the thing fount.